packages/xo-server/src/index.js

```diff
@@ -596,6 +596,8 @@ const setUpConsoleProxy = (webServer, xo) => {
 
     const [, id] = matches
     try {
+      const xapi = xo.getXapi(id, ['VM', 'VM-controller'])
```

Comment:
`xo.getXapiObject`?

Reply:
I don't have `XO` object.

Reply:
`getXapiObject` accepts ids.

Reply:
Is this the [implementation](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/xo-mixins/xen-servers.js#L481) of this method?

Reply:
My bad, you're right.

Maybe `xo.getXapiObject(xo.getObject(id, ['VM', 'VM-controller'])`?

Reply:
`xapi` is already used bellow. what's the point of using  `xo.getXapiObject(xo.getObject(id, ['VM', 'VM-controller'])` instead of `xo.getXapi(id, ['VM', 'VM-controller']).getObject(id)`

Reply:
In this case, your write it is the same, but only because the id of the XO object is also a valid id for the XAPI object, which is not always the case.

IMHO, the first syntax is better because always correct.

Feel free to update `getXapiObject` if you want it to support ids.

Reply:
so maybe `const xapi = xo.getXapi(xo.getObject(id, ['VM', 'VM-controller'])` to fix it one time for all because it's already used see https://github.com/vatesfr/xen-orchestra/pull/4867/files#diff-cdfc85d4b6db6988bde487b60b1907bdL629